Client/Ecosia/Migration/EcosiaImport.swift

```diff
@@ -43,6 +43,8 @@ final class EcosiaImport {
 
     private let migration = Migration()
     private var finished: ((Migration) -> ())?
+    private var progress: ((Double) -> ())?
```

Comment:
I wish we could use Combine for this. The list of retained callbacks is growing and that only makes me feel uneasy and unsafe. I'm fine with these for now but I think we can also start considering another pattern, Delegates come to mind but open for options.

Reply:
I see your point and would try avoiding stored blocks also but here it made sense to me. 

To mitigate your retain cycle concerns: AppDelegate-instance instantiates EcosiaImport but does not store a reference via property to it. So the instance is deinited after block execution. Also those blocks don't capture the EcosiaImport-instance, and we use [weak self] for AppDelegate-instance-capture (which lives the whole time anyhow) 
so I consider it safe this time - agree?

Reply:
I see your point and I think you are right in not adding safety (and complexity) if we know is not necessary.

The problem is that now we need to remember to keep it like that otherwise it is a time bomb. i.e.: if we try to move the migration out of the AppDelegate is now our responsibility to remember that it depends on this retain cycle to work properly.

For now let's keep it like that but let's keep an eye on it.